diff --git a/blog/admin.py b/blog/admin.py
index 120fab4..825a39e 100644
--- a/blog/admin.py
+++ b/blog/admin.py
@@ -1,6 +1,6 @@
 from django.contrib import admin
 
-from .models import Post
+from .models import Post, Comment
 from .forms import PostAdminForm
 
 
@@ -10,3 +10,10 @@ class PostAdmin(admin.ModelAdmin):
     readonly_fields = ('created', 'updated')
     prepopulated_fields = {'slug': ('title', )}
     form = PostAdminForm
+
+
+@admin.register(Comment)
+class PostCommentAdmin(admin.ModelAdmin):
+    list_display = ('name', 'email', 'active')
+    readonly_fields = ('created', 'updated')
+    
diff --git a/blog/forms.py b/blog/forms.py
index e2e41c2..751851e 100644
--- a/blog/forms.py
+++ b/blog/forms.py
@@ -2,7 +2,7 @@ from django import forms
 
 from pagedown.widgets import AdminPagedownWidget
 
-from .models import Post
+from .models import Post, Comment
 
 
 class PostAdminForm(forms.ModelForm):
@@ -29,3 +29,9 @@ class PostEmailForm(forms.Form):
     email = forms.EmailField()
     to = forms.EmailField()
     comment = forms.CharField(required=False, widget=forms.Textarea)
+
+
+class PostCommentForm(forms.ModelForm):
+    class Meta:
+        model = Comment
+        fields = ('name', 'email', 'body')
diff --git a/blog/models.py b/blog/models.py
index ec4469a..32a3d09 100644
--- a/blog/models.py
+++ b/blog/models.py
@@ -40,3 +40,13 @@ class Post(models.Model):
             self.publish.day,
             self.slug
         ])
+
+
+class Comment(models.Model):
+    post = models.ForeignKey(Post, on_delete=models.CASCADE, related_name="comments")
+    name = models.CharField(max_length=250)
+    email = models.EmailField()
+    body = models.TextField()
+    active = models.BooleanField(default=True)
+    created = models.DateTimeField(auto_now_add=True)
+    updated = models.DateTimeField(auto_now=True)
diff --git a/blog/templates/blog/post/post_detail.html b/blog/templates/blog/post/post_detail.html
index 6e130ac..3d24f0c 100644
--- a/blog/templates/blog/post/post_detail.html
+++ b/blog/templates/blog/post/post_detail.html
@@ -3,6 +3,8 @@
 {% extends 'base.html' %}
 {% load static %}
 {% load blog_tags %}
+{% load humanize %}
+{% load crispy_forms_tags %}
 
 {% block title %}
 {{ block.super }} - Blog - {{ post.title }}
@@ -33,16 +35,49 @@
 </div>
 <!-- Section 1 ends -->
 
-<!-- Section 2-->
+<!-- comments section -->
+
+<div id="comment_section" class="p-5">
+    <div class="container">
+        {% with comments.count as total_comments %}
+        {% if total_comments %}
+        <p class="text-right">{{ total_comments }} comment{{ total_comments|pluralize }}</p>
+        {% endif %}
+        {% endwith %}
+        <hr>
+
+        {% for comment in comments %}
+        <div class="{% cycle 'odd-comment' 'even-comment' %} p-2">
+            <span class="badge badge-success">#{{ forloop.counter }}</span>
+            <span class="badge badge-warning">{{ comment.name|title }}</span> said
+            <span class="badge badge-info">{{ comment.created|naturaltime }}</span>
+            <p>
+                {{ comment.body|linebreaks }}
+            </p>
+        </div>
+        {% endfor %}
+
+        <h3 class="mt-4">Leave a comment</h3>
+        <form method="post">
+            {% csrf_token %}
+            {{ form|crispy }}
+            <input type="submit" class="btn btn-dark" value="Submit">
+        </form>
+    </div>
+</div>
+
+<!-- comments section ends -->
+
+<!-- Section 2 -->
 <section id="quote" class="section-boundary" style="height: 300px;">
     <div class="container">
         <div class="row align-items-center" style="height: 300px;">
             <div class="col text-center">
-                <span class="display-4 d-none d-md-inline-block"><i class="fa fa-quote-right shadow"></i></span>  
+                <span class="display-4 d-none d-md-inline-block"><i class="fa fa-quote-right shadow"></i></span>
                 <span class="d-inline-block display-4" style="padding-right: 10px;padding-left: 10px;">
                     Do One Thing and Do It Well
                 </span>
-                <span class="display-4 d-none d-md-inline-block"><i class="fa fa-quote-right shadow"></i></span>  
+                <span class="display-4 d-none d-md-inline-block"><i class="fa fa-quote-right shadow"></i></span>
             </div>
         </div>
     </div>
diff --git a/blog/views.py b/blog/views.py
index 89afcdd..fb794cd 100644
--- a/blog/views.py
+++ b/blog/views.py
@@ -1,10 +1,11 @@
 import os
-from django.shortcuts import render, get_object_or_404
+from django.shortcuts import render, get_object_or_404, redirect
+from django.core.mail import send_mail
 from django.core.paginator import Paginator, PageNotAnInteger, EmptyPage
+from django.contrib import messages
 
 from .models import Post
-from .forms import PostEmailForm
-from django.core.mail import send_mail
+from .forms import PostEmailForm, PostCommentForm
 
 
 def post_list(request):
@@ -29,8 +30,38 @@ def post_detail(request, year, month, day, slug):
         Post, publish__year=year, publish__month=month, publish__day=day, slug=slug
     )
 
+    # Get all the active comments of this post
+    comments = post.comments.filter(active=True)
+
+    if request.method == "POST":
+        form = PostCommentForm(request.POST)
+        if form.is_valid():
+            # Create 'Comment' model instance backing the form
+            # but don't write to DB until for comment-post association
+            new_comment = form.save(commit=False)
+            # Associate post to the comment
+            new_comment.post = post
+            # Save to DB
+            new_comment.save()
+            # Message on successful submit
+            messages.add_message(
+                request,
+                messages.SUCCESS,
+                "Your comment is submitted"
+            )
+            # Redirect to this view
+            return redirect(to=post)
+    else:
+        form = PostCommentForm()
+
     return render(
-        request, "blog/post/post_detail.html", {"post": post, "active": "blog"}
+        request, "blog/post/post_detail.html",
+        {
+            "post": post,
+            "active": "blog",
+            'comments': comments,
+            'form': form,
+        }
     )
 
 
diff --git a/portfolio_project/settings.py b/portfolio_project/settings.py
index a8f0736..578bdac 100644
--- a/portfolio_project/settings.py
+++ b/portfolio_project/settings.py
@@ -38,6 +38,7 @@ INSTALLED_APPS = [
     'django.contrib.sessions',
     'django.contrib.messages',
     'django.contrib.staticfiles',
+    'django.contrib.humanize',
 
     # 3rd-party apps
     'pagedown.apps.PagedownConfig',
diff --git a/static/assets/css/styles.css b/static/assets/css/styles.css
index dea18cb..6518459 100644
--- a/static/assets/css/styles.css
+++ b/static/assets/css/styles.css
@@ -12,13 +12,17 @@
   --header-background-primary-color: var(--first-set-primary-color);
   --heading-underline-color: var(--first-set-primary-color);
   --list-tick-color: var(--first-set-primary-color);
-
+  --base-toast-background-color: var(--first-set-primary-color);
+  
   --second-set-primary-color: #d6d6d6;
   --second-set-secondary-color: #f0eaea;
   --title-underline-color: var(--second-set-primary-color);
   --about-background-color: var(--second-set-primary-color);
   --section-background-primary-color: var(--second-set-primary-color);
   --section-background-secondary-color: var(--second-set-secondary-color);
+  --comment-section-background-color: var(--second-set-secondary-color);
+  --odd-comment-background-color: var(--second-set-primary-color);
+  --even-comment-background-color: var(--comment-section-background-color);
 
   --third-set-primary-color: #f04216;
   --post-detail-link-color: var(--third-set-primary-color);
@@ -195,3 +199,21 @@ blockquote {
 blockquote > p {
     margin-bottom: 0;
 }
+
+#comment_section {
+  background-color: var(--comment-section-background-color);
+}
+
+.odd-comment {
+  background-color: var(--odd-comment-background-color);
+}
+
+.even-comment {
+  background-color: var(--even-comment-background-color);
+}
+
+.base-toast {
+  background-color: var(--base-toast-background-color)!important;
+  position: relative; 
+  min-height: 100px;
+}
diff --git a/templates/base.html b/templates/base.html
index bdcfb7e..6de66dd 100644
--- a/templates/base.html
+++ b/templates/base.html
@@ -18,6 +18,9 @@
     <!-- Navbar -->
     {% include 'navbar.html'%}
 
+    <!-- Messages -->
+    {% include 'message.html' %}
+
     <!-- Banner with overlay below the Navbar -->
     <div id="banner" class="header pt-5 p-md-none section-boundary">
         <div class="row h-100">
@@ -41,6 +44,13 @@
     <!-- Javascript libs -->
     <script src="{% static "assets/js/jquery.min.js" %}"></script>
     <script src="{% static "assets/bootstrap/js/bootstrap.min.js" %}"></script>
+
+    <!-- jquery for bootstrap toasts -->
+    <script>
+        $(document).ready(function() {
+            $('.toast').toast('show');
+        });
+    </script>
 </body>
 
 </html>
\ No newline at end of file
